### DISCUSSION TITLE
tokenizer不是右padding吗 为啥能直接在ppo中直接生成

### DISCUSSION THREAD

**Discussion by ciaoyizhen** [jingyaogong/minimind#597] [Q&A] [answered]
<img width="1656" height="817" alt="image" src="https://github.com/user-attachments/assets/64d88bf1-5c4e-426a-a6a2-e2090542b13f" />
我看tokenizer是右padding的，这不是会导致生成的结果不对吗？

<img width="1408" height="820" alt="image" src="https://github.com/user-attachments/assets/0f939e60-9dff-4022-8fe3-5d288b7aff9f" />
这里有大量的0  虽然有attention_mask  但是位置信息就不对了呀？  我看Qwen啥的 现在tokenizer都是左padding呀

**Comment by jingyaogong** [comment, marked as answer]
https://github.com/jingyaogong/minimind/blob/a9c56b20e94ffafae84092e0edff4aaca2f32d85/trainer/train_ppo.py#L303

已经是left padding了呀 🤔

**Comment by jingyaogong** [comment]
（之前的实现无问题）

目前已更新成显式 `left padding`：

https://github.com/jingyaogong/minimind/commit/11b962da062f13c0b643ce2aa90e3fe3722c1bcb#r173389609

https://github.com/jingyaogong/minimind/blob/11b962da062f13c0b643ce2aa90e3fe3722c1bcb/trainer/train_ppo.py#L126

**Comment by ciaoyizhen** [comment]
义父    哎呀 我没看到上面修改了哈哈哈哈
